lib/logreader.c

```diff
@@ -275,19 +291,26 @@ log_reader_update_watches(LogReader *self)
 
   log_reader_start_watches_if_stopped(self);
 
-  free_to_send = log_source_free_to_send(&self->super);
+  gboolean free_to_send = log_source_free_to_send(&self->super);
   if (!free_to_send)
     {
       log_reader_suspend_until_awoken(self);
       return;
     }
 
-  line_is_ready_in_buffer = log_proto_server_prepare(self->proto, &cond);
+  gboolean line_is_ready_in_buffer = log_proto_server_prepare(self->proto, &cond);
   if (self->immediate_check || line_is_ready_in_buffer)
     {
       log_reader_force_check_in_next_poll(self);
       return;
     }
+
+  if (!cond)
```

Comment:
Would it be possible to return a value from an enum?
It would be much more readable than using the default 0 value .